### DISCUSSION TITLE
The converted content of corejs@3 in preset-env is much less than the converted content of corejs@3 in runtime

### DISCUSSION THREAD

**Discussion by 1541157892** [babel/babel#15653] [Q&A]
Hello, I'm now running into something that's beyond my comprehension.
Why is the result of the same code using corejs@3 in preset-env less than half of the result obtained by using corejs@3 in runtime, all of which are processed by webpack using babel-loader.
Both of these questions have been bugging me for a while, and maybe it's not a big deal, but I haven't been able to get a relatively correct answer.

**Comment by nicolo-ribaudo** [comment]
I suspect it might be because global variables generare less code: you don't have to import them in every file where you need to use them, and you don't need checks like "is this really an array and thus I should use the polyfilled method, or it isn't?".

I would be happy to take a look at an example.

**Reply by 1541157892** [reply]
After reading your answer, I thought about it for a while, and I seem to have my own understanding.
Is it because if it is in the form of global variables, other dependencies will be used directly in the form of global variables if they need to be used.
And if it is in the form of local variables, then other dependencies need to do some repeated operations such as re-introduction when they are used. These operations lead to an increase in the code after the build

**Comment by zloirock** [comment]
Most likely it's because *entries* that injected via `runtime` contains all module dependencies, but *modules* injected via `preset-env` injected without unnecessary dependencies.

For example, you need only `AsyncIterator.from` method in an engine where it's not supported. `core-js-pure/full/async-iterator/from` entry that's injected via `runtime` contains dependency from `Promise` and polyfill for `Promise` will be injected too anyway. In the case of a modern engine where `Promise` is supported, `preset-env`  will inject only required modules like `core-js/modules/esnext.async-iterator.from`, and `Promise` polyfill will not be injected.

**Reply by 1541157892** [reply]
I didn't make any changes to .browserslistrc between the two builds.
Can I understand your answer as that the injection in preset-env is a member of the prototype, and the injection in runtime is the specific prototype and all its members?
